components/lib/datatable/TableBody.js

```diff
@@ -280,7 +280,14 @@ export const TableBody = React.memo(
             DomHandler.clearSelection();
             rangeRowIndex.current = allowCellSelection() ? event.rowIndex : event.index;
             let selectionInRange = selectRange(event);
-            let selection = isMultipleSelection() ? [...new Set([...(props.selection || []), ...selectionInRange])] : selectionInRange;
```

Comment:
@melloware I didn't understand this change. 'new Set' prevents duplicate items.

Reply:
If you revert it and look you will see it's allowing duplicate selections. Unlike Java in Ecmascript two identical objects are not  identical so it was allowing duplicate cells. 

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Look at the examples you will see adding two objects even if they have identical attributes will add it twice.

Reply:
Tomorrow I can send you before and after screenshots.

Reply:
In this part the rowData references are the same so duplication can be avoided. Exp;
```
var a = [{a: 1}, {b: 2}];
var b = [a[0]];

new Set([...a, ...b])
output: Set(2) {{a: 1}, {b: 2}}
```
Is there an issue or example reported about this?

Reply:
We can change it to just `let selection = selectRange(event);` and get rid of the Sets entirely